gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -26,14 +26,25 @@ module T::Private::Methods
   # in installed_hooks.
   @old_hooks = nil
 
-  @declaring_final_method = false
-
   ARG_NOT_PROVIDED = Object.new
   PROC_TYPE = Object.new
 
-  DeclarationBlock = Struct.new(:mod, :loc, :blk, :final)
+  DeclarationBlock = Struct.new(:mod, :loc, :blk, :final, :raw)
 
   def self.declare_sig(mod, arg, &blk)
+    # caller_depth is 2: 1 to get to our caller and 1 to get to sig()'s caller.
+    T::Private::DeclState.current.active_declaration = __declare_sig_internal(mod, arg, caller_depth: 2, &blk)
```

Comment:
I like the idea, but existing practice in `sorbet-runtime` appears to be a `caller_depth`-style approach -- possibly because the `caller_depth`-style enables you to delay the actual location gathering until you actually need to emit an error.

I suppose that logic doesn't apply here, since errors are unlikely and getting the location "early" isn't liable to cause performance issues.  (Compare with the implementation of casts, where we expect to not emit errors -- and therefore typically will not need location information -- so we're OK with `caller_depth`-style code.)

I'll fix this up.